core/src/main/java/edu/wpi/grip/core/operations/composite/PublishVideoOperation.java

```diff
@@ -31,16 +32,41 @@
  */
 public class PublishVideoOperation implements Operation {
 
+    public static final OperationDescription DESCRIPTION =
+            OperationDescription.builder()
+                    .name("Publish Video")
+                    .summary("Publish an M_JPEG stream to the dashboard.")
+                    .category(OperationDescription.Category.NETWORK)
+                    .icon(Icons.iconStream("publish-video"))
+                    .build();
+
     private final Logger logger = Logger.getLogger(PublishVideoOperation.class.getName());
 
     private static final int PORT = 1180;
     private static final byte[] MAGIC_NUMBER = {0x01, 0x00, 0x00, 0x00};
 
     private final Object imageLock = new Object();
     private final BytePointer imagePointer = new BytePointer();
-    private Optional<Thread> serverThread = Optional.empty();
     private volatile boolean connected = false;
-    private int numSteps = 0;
+    private Thread serverThread = null;
+    private static int numSteps = 0;
+
+    private final InputSocket<Mat> inputSocket;
+    private final InputSocket<Number> qualitySocket;
+
+    public PublishVideoOperation(InputSocket.Factory inputSocketFactory, OutputSocket.Factory outputSocketFactory) {
+        if (numSteps != 0) {
+            throw new IllegalStateException("Only one instance of PublishVideoOperation may exist");
+        }
+        this.inputSocket = inputSocketFactory.create(SocketHints.Inputs.createMatSocketHint("Image", false));
+        this.qualitySocket = inputSocketFactory.create(SocketHints.Inputs.createNumberSliderSocketHint("Quality", 80, 0, 100));
+        numSteps = numSteps + 1;
+        if (serverThread == null) {
```

Comment:
Why check to see if serverThread is null?  It should always be null in the constructor